front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -478,6 +481,29 @@ export function DocumentOrTableUploadOrEditModal({
                 )}
               </div>
 
+              {isTable &&
+                owner.flags.includes("use_app_for_header_detection") && (
```

Comment:
So for now this is the only path that triggers a header detection?

Reply:
connectors upserts will never use it for now if I understand correctly?

Reply:
Connector upserts for microsoft and google_drive uses header detection, when calling `upsertTableFromCsv()` - in both `spreadsheets.ts` , we have `useAppForHeaderDetection: true`. Not in notion, as databases are already well structured.

Reply:
For all workspaces?

Reply:
Only if feature flag is enabled. This is directly checked in api endpoint `front/pages/api/w/[wId]/data_sources/[dsId]/tables/csv.ts`, you must have the parameter set and the flag.

Reply:
ack